kafka_consumer/datadog_checks/kafka_consumer/client.py

```diff
@@ -186,91 +187,61 @@ def _get_consumer_offset_futures(self, consumer_groups):
-                            self.log.debug(
-                                "Partition %s skipped because the topic %s is not in the consumer_group.",
-                                partition,
-                                topic,
-                            )
-                            continue
-                        if (
-                            self.config._consumer_groups[consumer_group].get(topic)
-                            and partition not in self.config._consumer_groups[consumer_group][topic]
-                        ):
-                            self.log.debug(
-                                "Partition %s skipped because it is not defined in the consumer group for the topic %s",
-                                partition,
-                                topic,
-                            )
-                            continue
-
-                    topic_partition = TopicPartition(topic, partition)
-                    self.log.debug("TOPIC PARTITION: %s", topic_partition)
-                    yield topic_partition
-
-    def _get_regex_filtered_topic_partitions(self, consumer_group, topic, partitions):
-        for partition in partitions:
-            # Do a regex filtering here for consumer groups
-            for consumer_group_compiled_regex in self.config._consumer_groups_compiled_regex:
-                if not consumer_group_compiled_regex.match(consumer_group):
-                    return
-
-                consumer_group_topics_regex = self.config._consumer_groups_compiled_regex.get(
-                    consumer_group_compiled_regex
-                )
-
-                # If topics is empty, return all combinations of topic and partition
-                if not consumer_group_topics_regex:
-                    yield (topic, partition)
-
-                # Do a regex filtering here for topics
-                for topic_regex in consumer_group_topics_regex:
-                    if not topic_regex.match(topic):
-                        self.log.debug(
-                            "Partition %s skipped because the topic %s is not in the consumer_group.", partition, topic
-                        )
-                        continue
+                # No partitions specified means we collect all
+                if not topic_partitions:
+                    return set(partitions)
 
-                    if (
-                        consumer_group_topics_regex.get(topic_regex)
-                        and partition not in consumer_group_topics_regex[topic_regex]
-                    ):
-                        self.log.debug(
-                            "Partition %s skipped because it is not defined in the consumer group for the topic %s",
-                            partition,
-                            topic,
-                        )
-                        continue
+                partitions_to_collect.update(topic_partitions)
+
+        return partitions_to_collect.intersection(partitions)
```

Comment:
IIUC, if we return `partitions_to_collect`, we might be returning partitions that don't actually exist (it would return all partitions specified in the config for a given topic / topic regex). The intersection here matches the behavior of the original code of going through each partition and filtering it individually, as it is equivalent to (but probably less efficient than):

```python
return set(p for p in partitions if p in partitions_to_collect)
```